Add experimental loader

nicopap opened the pull request:
Add basic mesh loading for the fbx loader plugin.

- Write FBX asset loader plugin, derived from https://github.com/lo48576/fbx-viewer/ **Currently only supports single scene mesh only loading** skips all textures, materials, animations and what have you.
- Replace the `fbxcel_dom` loader by the bevy loader, so that we can test the bevy plugin
- Move the scene viewer to an example
- Remove some unused code. (can always checkout the code through git)

To test the asset loader, just run
```
cargo run --example scene_viewer
```
You can also pass as argument a path to a FBX file to load.

nicopap committed:
Add experimental loader

Currently broken yah, but we got some nice logs

nicopap committed:
Improve loading error reporting

HeavyRain266 commented:
Thanks, I'm currently working on CI tests for validating fbx files in `/scenes` filder

HeavyRain266 commented:
Instead of rewriting checker tool, put this bevy code in examples folder which will show how to work with fbx.

nicopap committed:
Make loader only load on 'A' keypress

nicopap committed:
Now something shows up on screen

- Add tangent computation through bevy_mikktspace
- Add a camera to the scene viewer, with controls, so that it's possible
  to see something
- Downscale the meshes since FBX is at 100x bevy's scale

nicopap committed:
Fix indices in mesh generation

nicopap committed:
Load multiple meshes from a file

nicopap committed:
Remove texture code

nicopap committed:
Use triangulate from utils

nicopap committed:
Move check_fbx to example file

nicopap committed:
Fix clippy lints

HeavyRain266 commented:
LGTM, thanks!
